phys/module_mp_ntu.F

```diff
@@ -277,25 +276,17 @@ SUBROUTINE NTU_INIT(PRES,z_at_q,RHOA,QDCN,QTCN,QCCN,QRCN,QNIN,   &
       DO J = JTS,JTF
          DO K = KTS,KTE
             DO I = ITS,ITF
-               IF (K.EQ.KTE) THEN
-                  ZATQ1(I,K,J) = z_at_q(I,K,J)-z_at_q(I,K-1,J)
-                  RHOA1(I,K,J) = 2.*RHOA(I,K-1,J)-RHOA(I,K-2,J)
-                  PRES1(I,K,J) = 2.*PRES(I,K-1,J)-PRES(I,K-2,J)
-               ELSE
-                  ZATQ1(I,K,J) = z_at_q(I,K+1,J)-z_at_q(I,K,J)
-                  RHOA1(I,K,J) = RHOA(I,K,J)
-                  PRES1(I,K,J) = PRES(I,K,J)
-               ENDIF
+               DZ(I,K,J) = (PHB(I,K,J)+PH(I,K,J))/G
             ENDDO
          ENDDO
       ENDDO
       DO J = JTS,JTF
-         DO K = KTS,KTE
+         DO K = KTS,KTE-1
             DO I = ITS,ITF
-               NK = KME-K+1
-               DZ8W(I,K,J) = ZATQ1(I,NK,J)
-               RHO(I,K,J) = RHOA1(I,NK,J)
-               P_PHY(I,K,J) = PRES1(I,NK,J)
+               NK = KME-K

```

Comment:
@tzuchin12 To be consistent with the loop indices, suggest to use KTE, instead of KME. Also, it seems you should define NK=KME-K+1, otherwise you will not have arrays defined at K=1.

Reply:
OKay, modified as suggested.  The values of KTE are different by 1 over the loops of init and main processes.

Reply:
@tzuchin12 Did you print and see that KTE and KME are different? They should be the same in general.

Reply:
I did. KTE and KME are the same in the ntu_init but KTE is less than KME by 1 in the main loop of microphysics processes.